################################
#Chlamydia trachomatis Project
################################

"""
Author: KaurS
Affiliation: Robert Koch Institute
Aim: A simple Snakemake workflow to process WGS data from clinical samples.
Run: snakemake  --cores 2   
generate rulegraph: snakemake --rulegraph 2> /dev/null | dot -T png > workflow.png
generate report: snakemake --report report.html
"""

import glob
import os
import pandas as pd


configfile: "config.yaml"
wdir = config["working_dir"]
SAMPLES, = glob_wildcards(wdir + "results/normalized/{sample}.norm_1.fastq.gz")

######### Target rules ###########

subpath = os.getcwd()


rule all:
	input:"results/multiqc/multiqc_report.html",
	      expand("results/var_call/filtered/{sample}.snps.table",sample=SAMPLES),
	      expand("results/var_call/filtered/{sample}.indels.table",sample=SAMPLES),
	      expand("results/mapping/addrg/{sample}.addrg.bam.bai", sample=SAMPLES)
	      		
	message: """--- Pipeline complete"""

######### Modules #############

#include: "rules/mapping.smk"
include: "rules/pre_processing.smk"
include: "rules/mapping_ct.smk"
include: "rules/variant_calling.smk"
include: "rules/variant_stats.smk"


